src/Database/Type/DateTimeType.php

```diff
@@ -267,7 +268,7 @@ public function marshal($value): ?DateTimeInterface
         if (is_array($value) && implode('', $value) === '') {
             return null;
         }
-        $value += ['hour' => 0, 'minute' => 0, 'second' => 0];
+        $value += ['hour' => 0, 'minute' => 0, 'second' => 0, 'microsecond' => 0];
```

Comment:
I can't find any use that requires changing.  FormHelper seems to just date a string and pass it to DateTime constructor.

Reply:
Since the multi selects for datetime inputs were removed from FormHelper in 4.x, no change in FormHelper is necessary.  

Personally I can't even think of a use case for user entered fractional seconds :slightly_smiling_face:.

Reply:
@ADmad Should I update the Validation::_getDateString()?  I'm not sure of the use case here.